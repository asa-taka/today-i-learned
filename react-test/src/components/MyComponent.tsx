import React, { useState } from 'react'

function CountButton() {
  const [count, setCount] = useState(0)
  return <button onClick={() => { setCount(s => s + 1)}}>{count}</button>
}

export default function MyComponent() {
  return (
    <div>
      <CountButton />
    </div>
  )
}
