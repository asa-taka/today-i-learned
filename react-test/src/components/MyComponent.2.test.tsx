import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import MyComponent from './MyComponent'

afterEach(cleanup)

test('MyComponent renders fine :)', () => {
  const { container } = render(<MyComponent />)
  expect(container).toMatchSnapshot()
})

test('MyComponent works fine :)', () => {
  const { container } = render(<MyComponent />)
  const button = container.querySelector('button')
  expect(button.textContent).toBe('0')
  fireEvent.click(button)
  expect(button.textContent).toBe('1')
})
