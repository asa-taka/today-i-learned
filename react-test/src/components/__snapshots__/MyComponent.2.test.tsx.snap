// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent renders fine :) 1`] = `
<div>
  <div>
    <button>
      0
    </button>
  </div>
</div>
`;
